File: Solutions/Count_Good_Nodes_in_Binary_Tree/1448. Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode,val=-10000) -> int:
        return self.goodNodes(root.left,max(val,root.val))+self.goodNodes(root.right,max(val,root.val))+(val<=root.val) if root else 0

File: Solutions/Count_Good_Nodes_in_Binary_Tree/1448. Count Good Nodes in Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
import "math"
func goodNodes(root *TreeNode) int {
    if root==nil{
        return 0;
    }
    res:=0
    foo(&res,root,math.MinInt32)
    return res;
}
func foo(res *int,root *TreeNode,val int){
    if root!=nil{
        if val<=root.Val{
            *res+=1
        }
        foo(res,root.Left,max(root.Val,val))
        foo(res,root.Right,max(root.Val,val))
    }
}
func max(a,b int)int{
    if a<b{
        return b
    }
    return a
}